runtime/near-vm/engine-universal/src/code_memory.rs

```diff
@@ -143,51 +132,20 @@ impl CodeMemory {
 
     /// Calculates the allocation size of the given compiled function.
     fn function_allocation_size(func: FunctionBodyRef<'_>) -> usize {
-        match &func.unwind_info {
-            Some(CompiledFunctionUnwindInfoRef::WindowsX64(info)) => {
-                // Windows unwind information is required to be emitted into code memory
-                // This is because it must be a positive relative offset from the start of the memory
-                // Account for necessary unwind information alignment padding (32-bit alignment)
-                ((func.body.len() + 3) & !3) + info.len()
-            }
-            _ => func.body.len(),
-        }
+        func.body.len()
     }
 
     /// Copies the data of the compiled function to the given buffer.
     ///
     /// This will also add the function to the current function table.
-    fn copy_function<'a>(
-        registry: &mut UnwindRegistry,
-        func: FunctionBodyRef<'_>,
-        buf: &'a mut [u8],
-    ) -> &'a mut [VMFunctionBody] {
+    fn copy_function<'a>(func: FunctionBodyRef<'_>, buf: &'a mut [u8]) -> &'a mut [VMFunctionBody] {
         assert_eq!(buf.as_ptr() as usize % ARCH_FUNCTION_ALIGNMENT, 0);
 
         let func_len = func.body.len();
 
-        let (body, remainder) = buf.split_at_mut(func_len);
```

Comment:
What would this guard against? It seems like it would be perfectly valid for a caller to allocate a larger buffer and pass that in, if they expect to write more stuff in one go, or in case they allocate at a granularity of a page. Conversely, this function can't avoid this `split_at_mut`, as `copy_from_slice` requires slices of equal length.

Reply:
Makes sense, I didn’t think of the fact that it could be wanted behavior from the user!